North-Seattle-College/ad440-winter2022-thursday-repo
# Develop automation script to create the service that will host the ML model

### Closes issue #17

### Overview
- This pull request adds two files: `create_sagemaker_nbi.py` and `sagemaker_nbi.json`
- Together, they create a SageMaker notebook instance that can be used for data science/machine learning projects
- Note: Running the Python script will attempt to create the notebook instance, but will fail if user is not authenticated or has insufficient IAM permissions

### Prerequisites
- Python Boto3 installed (python -m pip install boto3)
- AWS CLI and proper basic configuration

### To Test
1. Open branch/download the scripts: `create_sagemaker_nbi.py` and `sagemaker_nbi.json`
2. Run create_sagemaker_nbi.py and input initials when prompted
3. Script will run for some time, and then return an error to the console: `Waiter StackCreateComplete failed`
     - Students do not have the required permissions to create this resource in our AWS account
4. Verify that a stack create was attempted by going to CloudFormation in the AWS Console (you should see a failed stack with your initials - "ROLLBACK_COMPLETE")
![notebook_instance](https://user-images.githubusercontent.com/45498644/151924268-c4654bb2-668a-4704-a72d-c32ce94a9ff4.png)

5. Delete your stack from the AWS Console for housekeeping purposes.

### Time Spent
| Activity | Estimate | Actual Time Spent |
| ------ | ------ | ------ |
| Research SageMaker functionality | 2 hours | 3 hours |
| Write automation script | 3 hours | 2 hours |
| Test automation script | 1 hour | 1 hour |
| Write documentation | 1 hour | 1 hour |
| Pull request and wrap-up | 1 hour | 30 mins
